servers/netserver/src/bpf/bpf.cpp

```
@@ -0,0 +1,125 @@
+				bpf_log_op("A (0x%x) &= k (0x%x) = 0x%x", A, inst.k, A & inst.k);
+				A &= inst.k;
+				break;
+			}
+			case Op::ALU_MUL_K: {
+				bpf_log_op("A (0x%x) *= k (0x%x) = 0x%x", A, inst.k, A * inst.k);
+				A *= inst.k;
+				break;
+			}
+			case Op::JMP_JEQ_K: {
+				bpf_log_op("PC += 0x%02x if A == K (0x%x == 0x%x) else 0x%02x (0x%02x)", inst.jt, A, inst.k, inst.jf, (A == inst.k) ? inst.jt : inst.jf);
+				pc += (A == inst.k) ? inst.jt : inst.jf;
+				break;
+			}
+			case Op::JMP_JSET_K: {
+				bpf_log_op("PC += 0x%02x if A & k (0x%x & 0x%x) else 0x%02x (0x%02x)", inst.jt, A, inst.k, inst.jf, (A & inst.k) ? inst.jt : inst.jf);
+				pc += (A & inst.k) ? inst.jt : inst.jf;
+				break;
+			}
+			case Op::LDX_W_IMM: {
+				bpf_log_op("X <- k (0x%02x)", inst.k);
+				X = inst.k;
+				break;
+			}
+			case Op::LD_B_IND: {
+				auto val = load.template operator()<uint8_t>(X + inst.k);
+				bpf_log_op("A <- P[X+k:1 (0x%02x + 0x%02x)] (0x%hx)", X, inst.k, val);
+				A = val;
+				break;
+			}
+			case Op::LD_H_ABS: {
+				auto val = load.template operator()<uint16_t>(inst.k);
+				bpf_log_op("A <- P[k:2 (0x%02x)] = 0x%hx", inst.k, val);
+				A = val;
+				break;
+			}
+			case Op::LD_H_IND: {
+				auto val = load.template operator()<uint16_t>(X + inst.k);
+				bpf_log_op("A <- P[X+k:2 (0x%02x + 0x%02x)] (0x%hx)", X, inst.k, val);
+				A = val;
+				break;
+			}
+			case Op::LD_W_IND: {
+				auto val = load.template operator()<uint32_t>(X + inst.k);
+				bpf_log_op("A <- P[X+k:4 (0x%02x + 0x%02x)] (0x%hx)", X, inst.k, val);
+				A = val;
+				break;
+			}
+			case Op::MISC_TAX: {
+				bpf_log_op("X <- A (0x%02x)", A);
+				X = A;
+				break;
+			}
+			case Op::RET_K: {
+				bpf_log_op("RET k (0x%02x)", inst.k);
+				return inst.k;
+			}
+			default:
+				printf("netserver: unhandled BPF instruction 0x%02x\n", inst.code);
+				bpf_log_op("{ 0x%02x, %.2u, %.2u, 0x%08x }\n", inst.code, inst.jt, inst.jf, inst.k);
+				assert(!"unhandled BPF instruction");
```

We should probably not be asserting on something like this

---

Given that the BPF support in this PR is only covering a part of the instruction set, failing catastrophically instead of silently failing seemed more appropriate.

---

We should not fail "silently", but returning an error or something is probably more appropriate imo